packages/fuels-signers/src/lib.rs

```diff
@@ -202,81 +217,108 @@ mod tests {
 
         let wallet_1_spendable_resources =
             wallet_1.get_spendable_resources(BASE_ASSET_ID, 1).await?;
-        let wallet_2_spendable_resources = wallet_2
-            .get_spendable_resources(BASE_ASSET_ID, amount + 1)
-            .await?;
+        let wallet_2_spendable_resources =
+            wallet_2.get_spendable_resources(BASE_ASSET_ID, 1).await?;
         let wallet_1_all_coins = wallet_1.get_coins(BASE_ASSET_ID).await?;
         let wallet_2_all_coins = wallet_2.get_coins(BASE_ASSET_ID).await?;
 
         // wallet_1 has now only one spent coin and one not spent(the remaining not sent coins)
         assert_eq!(wallet_1_spendable_resources.len(), 1);
         assert_eq!(wallet_1_all_coins.len(), 2);
-        assert_eq!(wallet_2_spendable_resources.len(), 2);
-        // Check that wallet two now has two coins.
-        assert_eq!(wallet_2_all_coins.len(), 2);
+        // Check that wallet two now has a coin.
+        assert_eq!(wallet_2_all_coins.len(), 1);
+        assert_eq!(wallet_2_spendable_resources.len(), 1);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn transfer_more_than_owned() -> fuels_types::errors::Result<()> {
+        const AMOUNT: u64 = 1000000;
+        let (wallet_1, wallet_2) = setup_transfer_test(AMOUNT).await;
 
         // Transferring more than balance should fail.
         let response = wallet_1
             .transfer(
                 wallet_2.address(),
-                2000000,
+                AMOUNT * 2,
                 Default::default(),
                 TxParameters::default(),
             )
             .await;
 
         assert!(response.is_err());
         let wallet_2_coins = wallet_2.get_coins(BASE_ASSET_ID).await?;
-        assert_eq!(wallet_2_coins.len(), 2); // Not changed
+        assert_eq!(wallet_2_coins.len(), 0);
         Ok(())
     }
 
     #[tokio::test]
     async fn transfer_coins_with_change() -> fuels_types::errors::Result<()> {
-        // Setup two sets of coins, one for each wallet, each containing 1 coin with 5 amounts each.
+        const AMOUNT: u64 = 5;
+        let (wallet_1, wallet_2) = setup_transfer_test(AMOUNT).await;
+
+        // Transfer 2 from wallet 1 to wallet 2.
+        const SEND_AMOUNT: u64 = 2;
+        let _receipts = wallet_1
+            .transfer(
+                wallet_2.address(),
+                SEND_AMOUNT,
+                BASE_ASSET_ID,
+                TxParameters::default(),
+            )
+            .await?;
+
+        let wallet_1_final_coins = wallet_1.get_spendable_resources(BASE_ASSET_ID, 1).await?;
```

Comment:
I don't have an answer to this yet, and I'm not even sure this could be a source of confusion, but one thing came to mind when reading `wallet_1.get_spendable_resources(BASE_ASSET_ID, 1)`: two methods with the same name (`get_spendable_resources`) that run the same operation (filtering) on the same data (resources). One takes a neat `ResourceFilter`, the other... two separate args. It would be neat if they had the same signature, but I know `ResourceFilter` takes the address, and `get_spendable_resources` bound to the wallet doesn't take an address. How could we unify these two APIs?

Reply:
I hoped to make the changes less annoying by retaining parts of the old API. But the point is valid; name and behavior are the same, which could be confusing. 
How about extracting the address from the filter struct?

Reply:
Yeah, I think that would be the best solution. Let's do it!

Reply:
It would solve the signature issue, but I feel kinda queasy about it. 

That would make the original `get_spendable_resources` on the Provider look like this:
```rust
    pub async fn get_spendable_resources(
        &self,
        address: Bech32Address,
        filter: ResourceFilter,
    ) -> ProviderResult<Vec<Resource>> {
```
But the `address` is just as much part of the filtering as `filter` is. I'm ok with additional arguments alongside `filter` but only if they aren't directly part of the filtering. Like if we had `retry_count` for retrying the query or something like that.

I'm for either leaving it as is -- a shortcut to get wallet-owned spendable resources OR going full bananas with it.

We no longer have `ResourceFilters` but a `ResourceQuery` which you can run.

So searching for resources might look like this:
```rust
let resource_query = provider.resource_query().from(from).amount(100); 
resource_query.run();
```

Then you can use the same thing to populate the `from` field from the wallet:
```rust
let query = wallet.get_spendable_resources();
query.amount(100);
let result = query.run().await;
```
where `wallet` will populate `from` before returning it from `get_spendable_resources`.

The query object inside would have to hold the provider either by value or reference, whatever we see is more ergonomic.

Reply:
> But the address is just as much part of the filtering as filter is.

Yeah, I've been trying to convince myself that this would be okay... but, nah, you're right, feels clunky. The address is a filtering predicate. 

> I'm for either leaving it as is -- a shortcut to get wallet-owned spendable resources 

Yup, I'd say let's keep as is and call the one bound to the wallet a shortcut.